### ISSUE TITLE
org.shipkit.gradle-plugin releases even if there are no changes in binaries

### ISSUE THREAD

**Issue by szpak** [mockito/shipkit#419]
### Problem

Preventing the release happening when there are not changes in the binary files is not currently implemented for 'org.shipkit.gradle-plugin' users. It is only implemented for 'org.shipkit.java' users.

### Implementation

The plugin that compares publications is called "ComparePublicationsPlugin" and it is applied when 'org.shipkit.java' is in use. Let's figure out how can we make 'org.shipkit.gradle-plugin' leverage compare publications plugin.

### Original report from @szpak

I've just [tweaked](https://github.com/mockito/shipkit/commit/1c5a8727dfbc01f644db4c937c0dbb7b256ae20e) release notes and the new version has been released (as the publications was [not equal](https://travis-ci.org/mockito/shipkit/builds/276207813#L1551)). Is it the expected behavior?

**Comment by wwilk** [comment]
It's not the expected behaviour. We compare poms and sources jars there and docs/ should not be a part of this. I will investigate it today, thanks for opening this issue!

BTW we have a strategy to create pull requests even for small changes like this one, and merge it only when someone else approves it or if it stays too long as open and is blocking us.

**Comment by szczepiq** [comment]
I saw this in the logs:

```
[assertReleaseNeeded]   Publication comparison was skipped (no comparison result files found).
```

I will look into this

**Comment by szczepiq** [comment]
We currently don't support skipping publications if binaries are the same for Gradle plugins. Our "org.shipkit.java" plugin supports it, but the "org.shipkit.gradle" does not. It should not be too difficult to add this. Who wants to give it a stab? :) The plugin that compares publications is called "ComparePublicationsPlugin"

**Comment by szczepiq** [comment]
Updated the description. @szpak, do you want to take a stab at it?

**Comment by wwilk** [comment]
OK I'm starting to work on this, since lack of it seems to create an issue #443

**Comment by wwilk** [comment]
@szpak Please let me know if you already started working on it

**Comment by szczepiq** [comment]
@wwilk, it seems the issue is not yet fixed, even though we added compare publications plugin. I saw the PR opened for a documentation-only change (markdown files). Pull request: https://github.com/mockito/shipkit/pull/492

**Comment by wwilk** [comment]
Yeah it seems that ComparePublications finds differences in pom when there is none. I'll investigate

**Comment by szpak** [comment]
@wwilk No, I haven't started working on it yet.

**Comment by wwilk** [comment]
Fixed in version 0.9.139
